resources/js/api/cities.ts

```diff
@@ -0,0 +1,53 @@
+const API_BASE_URL = "/api/cities";
+
+function makeRequest<T>(
+  method: string,
+  url: string,
+  body: any = null,
+): Promise<T> {
```

Comment:
```suggestion
function makeRequest<K, T>(
  method: string,
  url: string,
  body: K = null,
): Promise<T> {
```

Maybe it's a good idea, so we check that the body is what we are expecting